fuse_core/src/uuid.cpp

```diff
@@ -96,6 +96,13 @@ UUID generate(const std::string& namespace_string, const ros::Time& stamp, const
   return generate(namespace_string, buffer.data(), buffer.size());
 }
 
+UUID generate(const std::string& namespace_string, const uint64_t& user_id)
+{
+  ros::Time id_as_time;
+  id_as_time.fromNSec(user_id);
+  return generate(namespace_string, id_as_time);
+}
```

Comment:
Converting the user id into a timestamp just to extract the raw bytes back out seems inefficient
```suggestion
UUID generate(const std::string& namespace_string, const uint64_t& user_id)
{
  return generate(namespace_string, reinterpret_cast<const unsigned char*>(&user_id), sizeof(user_id));
}
```